pkg/ccl/changefeedccl/cdcevent/event.go

```diff
@@ -0,0 +1,491 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package cdcevent
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/lease"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
+	"github.com/cockroachdb/cockroach/pkg/sql/row"
+	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/cache"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/iterutil"
+	"github.com/cockroachdb/errors"
+)
+
+// EventSource identifies the source of the event.
```

Comment:
I'm not in love with this as an interface. Effectively all of the methods are constants. The ones that aren't change very rarely. How would you feel about making this a struct? Like cdcevent.Metadata or Info or something? I also find the name of it to be pretty bad. It's not a source with these methods. 

I don't think you'll have to change much of anything to export the eventDescriptor struct as it is. I think it'd even be fine to embed that newly exported struct and leave all of its fields private as they currently are. You'd need to change almost nothing.

Reply:
> I'm not in love with this as an interface. Effectively all of the methods are constants. The ones that aren't change very rarely. How would you feel about making this a struct? Like cdcevent.Metadata or Info or something? I also find the name of it to be pretty bad. It's not a source with these methods.
> 
> I don't think you'll have to change much of anything to export the eventDescriptor struct as it is. I think it'd even be fine to embed that newly exported struct and leave all of its fields private as they currently are. You'd need to change almost nothing.

Name -- I don't mind -- it was the source for the EventRow (or Row) as it's being renamed.
It's alright to change things. Changing it to struct makes sense.  
I think the reason for it being an interface was an implementation, I think in cloud storage sink test, that mucked with version... But... that's even easier to accomplish with a struct containing public (constant) members.